Android/Shared/src/main/java/com/bentley/sample/shared/SampleITMApplication.kt

```diff
@@ -77,6 +79,44 @@ open class SampleITMApplication(context: Context, attachWebViewLogger: Boolean,
                 throw ex
             }
         }
+
+        coMessenger.registerMessageHandler("firstRenderStarted") {
+            logger.log(ITMLogger.Severity.Debug, "Received firstRenderStarted")
+        }
+        coMessenger.registerMessageHandler("firstRenderFinished") {
+            logger.log(ITMLogger.Severity.Debug, "Received firstRenderFinished")
+        }
+    }
+
+    /**
+     * Checks for values in configData with a prefix of "ITMSAMPLE_ACTION_" and constructs a [JsonObject]
+     * using everything after ITMSAMPLE_ACTION_ as the key and the values from configData.
+     * @return A [JsonObject] containing all the ITMSAMPLE_ACTION_ prefixed values from configData.
+     */
+    protected open fun getActionsFromConfigData(): JsonObject {
+        val actions = JsonObject()
+        val configData = this.configData ?: return actions
+        for (key in configData.names()) {
+            val shortKey = key.removePrefix("ITMSAMPLE_ACTION_")
+            if (shortKey.length == key.length) { continue }
```

Comment:
Shouldn't this be: `if (shortKey.length == 0)`?

Reply:
No; if the prefix isn't there, it returns the original string.

Reply:
Sorry, I misunderstood, it should be fine what you have.